src/plasma/plasma_manager.cc

```diff
@@ -589,13 +592,14 @@ void send_queued_request(event_loop *loop,
     break;
   case MessageType_PlasmaDataReply:
     LOG_DEBUG("Transferring object to manager");
-    if (conn->cursor == 0) {
-      /* If the cursor is zero, we haven't sent any requests for this object
+    if (ClientConnection_request_finished(conn)) {
```

Comment:
We used to be checking `conn->cursor ==  0`, and now we're checking `conn->cursor == -1`. Is the danger that if we only use `0` instead of `0` and `-1` then we may accidentally go through this code block twice if `write_object_chunk` fails to write something?

Or did it just seem cleaner to do it this way?

Reply:
Yeah, I was worried about the case you mentioned happening. I guess the chances are small, but it seems cleaner to do it this way anyway.